source/chapter7/graphics/render_scene.cpp

```diff
@@ -2143,7 +2143,7 @@ void RenderScene::upload_gpu_data( UploadGpuDataContext& context ) {
         for ( u32 i = 0; i < k_num_lights; ++i ) {
             const SortedLight& light = sorted_lights[ i ];
 
-            if ( ( light.projected_z >= bin_min && light.projected_z <= bin_max ) ||
+            if ( ( light.projected_z_min <= bin_min && light.projected_z_max >= bin_max ) ||
```

Comment:
Not sure this is correct, as this would include lights that have their min/max depth bounds outside the bin bounds. It could well be that this line is not needed though and that we just need to check the min/max z against the bin bounds. I haven't run the code yet, I'll verify more thoroughly later.